save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/modal/ModalBuilder.kt

```diff
@@ -271,3 +256,69 @@ fun ChildrenBuilder.modalBuilder(
         }
     }
 }
+
+/**
+ * Function to create modals with bootstrap styles for upload avatars.
+ *
+ * @param title title of the modal that will be shown in top-left corner
+ * @param onCloseButtonPressed callback that will be applied to `X` button in the top-right corner
+ * @param buttonBuilder lambda that generates several buttons, must contain either [button] or [buttonBuilder]
+ * @param isOpen
+ */
+fun ChildrenBuilder.modalAvatarBuilder(
+    isOpen: Boolean,
+    title: String,
+    onCloseButtonPressed: (() -> Unit)?,
+    buttonBuilder: ChildrenBuilder.() -> Unit,
+) {
+    modal { props ->
+        props.isOpen = isOpen
+        props.style = mediumTransparentModalStyle
+        div {
+            className = ClassName("modal-dialog")
+            div {
+                className = ClassName("modal-content")
+                onCloseFun(
+                    title,
+                    onCloseButtonPressed,
+                )
+                div {
+                    className = ClassName("justify-content-center modal-footer")
+                    buttonBuilder()
+                }
+                div {
+                    className = ClassName("modal-body")
+                }
+            }
+        }
+    }
+}
+
+/**
+ * @param title
+ * @param onCloseButtonPressed
```

Comment:
Please document the use-case (why and when) `onCloseButtonPressed` may be `null`.

Why is it optional?

Reply:
`onCloseFun` still has `onCloseButtonPressed` as nullable